Source/Core/VideoCommon/Fifo.cpp

```diff
@@ -496,35 +456,89 @@ int Fifo_Update(int ticks)
 {
 	const SConfig& param = SConfig::GetInstance();
 
-	if (ticks == 0)
+	if (!param.bCPUThread || g_use_deterministic_gpu_thread)
```

Comment:
The if and else cases should each be their own function with a self-documenting name indicating what they do (the fact the if case is just shuffled code reinforces this).